scripts/release.py

```diff
@@ -402,6 +458,38 @@ def request_doc_update(github_token, repo, release):
 
 def build_parser():
     parser = argparse.ArgumentParser(description='Publish RPCO tag.')
+    # Mandatory args
+    parser.add_argument(
+        '--tag', type=Tag, required=True, help='Name of new tag.'
+    )
+    existing = parser.add_mutually_exclusive_group(required=True)
+    existing.add_argument(
+        '--commit', help='Reference to the commit to tag.'
+    )
+    existing.add_argument(
+        '--existing-release', default=False, action='store_true'
+    )
+    # Recommended args
+    parser.add_argument(
+        '--future-tag', type=Tag,
+        help='The version number for the next cycle'
+    )
+    # Convenience args that defaults to ENV vars
+    parser.add_argument(
+        '--branch', help='Branch to update'
```

Comment:
For example,
```python
chk_devel_version(rpco_repo, args.branch, args.tag_str)
```
is used to confirm that the tag specified is the same set in the branch vars files. Following that the tag is created using the commit:
```python
rpco_tag = rpco_repo.create_tag(                                        
    tag_str=args.tag_str, commit=args.commit, message=tag_message       
)
```
It seems there are a couple of assumptions being made:
- the commit can be reached from the branch
- the vars files at the HEAD of the branch match the commit

I've not traced the use of branch and commit through the whole script but, at least based on this example, their current use allows the potential for mistakes.